build/automation/lib/file.mk

```
@@ -24,13 +24,14 @@ file-replace-variables: ### Replace all placholders in given file - mandatory: F
 		value=$$(echo $$(eval echo "\$$$$key"))
 		if [ -z "$$value" ]; then
 			echo "WARNING: Variable $$key has no value in '$$(echo $(FILE) | sed "s;$(PROJECT_DIR);;g")'"
-		else
-			# Replace `${VARIABLE_TO_REPLACE}`
-			sed -i "s;\$${$${key}$${suffix}};$${value//&/\\&};g" $(FILE)
-			# Replace `$VARIABLE_TO_REPLACE`
-			sed -i "s;\$$$${key}$${suffix};$${value//&/\\&};g" $(FILE)
-			# Replace `VARIABLE_TO_REPLACE`
-			sed -i "s;$${key}$${suffix};$${value//&/\\&};g" $(FILE)
+		fi
+		if [ ! -z "$$value" ] || [[ "$(REPLACE_NO_VALUE)" =~ ^(true|yes|y|on|1|TRUE|YES|Y|ON)$$ ]]; then
+		# Replace `${VARIABLE_TO_REPLACE}`
+		sed -i "s;\$${$${key}$${suffix}};$${value//&/\\&};g" $(FILE)
+		# Replace `$VARIABLE_TO_REPLACE`
+		sed -i "s;\$$$${key}$${suffix};$${value//&/\\&};g" $(FILE)
+		# Replace `VARIABLE_TO_REPLACE`
+		sed -i "s;$${key}$${suffix};$${value//&/\\&};g" $(FILE)
 		fi
```

Not convinced these comments are adding anything, also the indentation is off
```suggestion
		if [ ! -z "$$value" ] || [[ "$(REPLACE_NO_VALUE)" =~ ^(true|yes|y|on|1|TRUE|YES|Y|ON)$$ ]]; then
		  sed -i "s;\$${$${key}$${suffix}};$${value//&/\\&};g" $(FILE)
  		sed -i "s;\$$$${key}$${suffix};$${value//&/\\&};g" $(FILE)
  		sed -i "s;$${key}$${suffix};$${value//&/\\&};g" $(FILE)
		fi
```